sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -269,7 +269,7 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
     } else {
       val (useTempCheckpointLocation, recoverFromCheckpointLocation) =
         if (source == "console") {
-          (true, false)
+          (true, true)
```

Comment:
Is there any source that uses `recoverFromCheckpointLocation` disabled? What's the use case if any?

Remove  `recoverFromCheckpointLocation` here as it's always `true` and make it explicit.

The JIRA issue is to fix the exception followed by cleaning the code that was needed in the past.

Reply:
Good point @jaceklaskowski updated.thanks